### ISSUE TITLE
CMake not able to find MathGL

### ISSUE THREAD

**Issue by MarcoCallari** [microsoft/vcpkg#22264]
**Describe the bug**
find_package does not find the MathGL library. I have installed the package mathgl:x64-windows. I have other packages installed with vcpkg and none of them cause any trouble.

**Environment**
- OS: Windows 10
- Compiler: MSVC 19.29.30138
-CMake version 3.22.0 
-Commit 95d4d067f05b36225b292bf1c3156a7a20c913c0 of the vcpkg repo

**To Reproduce**
1. ./vcpkg.exe install mathgl:x64-windows
2. Try to find the package as instructed in vcpkg:
In a CMakeLists.txt file:
find_package(mathgl CONFIG REQUIRED)
2. cmake -DCMAKE_TOOLCHAIN_FILE="[path]" -DVCPKG_TARGET_TRIPLET="x64-windows" ..

**Expected behavior**
CMake should be able to find the package. On my windows 10 machine, it does not. I also have other packages installed with vcpkg (such as eigen3 and gsl) which CMake is able to find without causing any problem.

**Failure logs** 
 (_find_package):
  Could not find a package configuration file provided by "mathgl" with any
  of the following names:

    mathglConfig.cmake
    mathgl-config.cmake

  Add the installation prefix of "mathgl" to CMAKE_PREFIX_PATH or set
  "mathgl_DIR" to a directory containing one of the above files.  If "mathgl"
  provides a separate development package or SDK, be sure it has been
  installed.

**Additional context**

**Comment by FrankXie05** [comment]
@MarcoCallari Could you please provide your CMakeLists.txt?

**Comment by MarcoCallari** [comment]
Sure thing. The minimal reproducible example is:
CMakeLists.txt
```
project(mathgltest)
cmake_minimum_required(VERSION 3.22)
find_package(mathgl CONFIG REQUIRED)
add_executable(test main.cpp)
target_link_libraries(test mgl mgl-static)
```

Where main.cpp is simply
```
int main(void) { return 0; }
```

**Comment by FrankXie05** [comment]
@MarcoCallari Sorry for the late reply, I think this is a bug, I will submit a pr to solve it immediately. :)

**Comment by nil-is-lin** [comment]
Is there still no solution to this problem?

**Comment by Adela0814** [comment]
I have fixed the usage problem in the PR #22634, this PR has other issues that have not been resolved but does not affect the usage

**Comment by YiHui-Liu** [comment]
> I have fixed the usage problem in the PR #22634, this PR has other issues that have not been resolved but does not affect the usage.

It not works now.

**Comment by dg0yt** [comment]
The only supported usage with #28311 will be:
~~~
    find_package(MathGL2 CONFIG REQUIRED)
    target_link_libraries(main PRIVATE $<IF:$<TARGET_EXISTS:mgl>,mgl,mgl-static>)
~~~
This is exported CMake config which properly works across triplets and build types.
